platform/linux-generic/_ishm.c

```diff
@@ -1436,15 +1436,23 @@ int _odp_ishm_cleanup_files(const char *dirpath)
 	return 0;
 }
 
-int _odp_ishm_init_global(void)
+int _odp_ishm_init_global(const odp_init_t *init)
 {
 	void *addr;
 	void *spce_addr;
 	int i;
 	uid_t uid;
 	char *hp_dir = odp_global_data.hugepage_info.default_huge_page_dir;
 	uint64_t align;
+	uint64_t max_memory = ODP_CONFIG_ISHM_VA_PREALLOC_SZ;
+	uint64_t internal   = ODP_CONFIG_ISHM_VA_PREALLOC_SZ / 8;
```

Comment:
'internal' has to be also adjusted. Or you can get overflow at line 1455.

Reply:
if (init && init->shm.max_memory)
		max_memory = init->shm.max_memory + internal;

line 1455 does not underflow even with application requested init->shm.max_memory value:
shm_max_size = init->shm.max_memory + internal - internal;

Reply:
Internal is for ODP implementation internal allocations. E.g. if application passes 1MB as init->shm.max_memory, we need to reserve much more as ODP internally needs SHM for various tables, etc. Application's max_memory just says how much it will itself request.